src/pkg/cli/compose/fixup.go

```diff
@@ -70,7 +72,24 @@ func FixupServices(ctx context.Context, provider client.Provider, project *types
 	for _, svccfg := range project.Services {
 		// Upload the build context, if any; TODO: parallelize
 		if svccfg.Build != nil {
-			// Pack the build context into a tarball and upload
+			// Because of normalization, Dockerfile is always set to "Dockerfile" even if it was not specified in the compose file.
+			if svccfg.Build.Dockerfile != "" {
+				// Check if the dockerfile exists
+				dockerfilePath := filepath.Join(svccfg.Build.Context, svccfg.Build.Dockerfile)
+				if _, err := os.Stat(dockerfilePath); err != nil {
+					// In this case we know that the dockerfile is not in the location the compose file specifies,
+					// so can assume that the dockerfile has been normalized to the default "Dockerfile".
+					if svccfg.Build.Dockerfile != "Dockerfile" {
+						// An explicit Dockerfile was specified, but it does not exist.
+						return fmt.Errorf("service %q: dockerfile not found: %w", svccfg.Name, ErrDockerfileNotFound)
+					}
+
+					// Undo normalization
+					svccfg.Build.Dockerfile = ""
```

Comment:
we risk "swalling" the `err` in this branch so we should add a `term.Debug` here.